src/routers/resource_router.py

```
@@ -485,13 +486,35 @@ def register_resource(
 
         return register_resource
 
-    def create_resource(self, session: Session, resource_create_instance: SQLModel):
+    def create_resource(
+        self,
+        session: Session,
+        resource_create_instance: SQLModel,
+        user: KeycloakUser = Depends(get_user_or_raise),
+    ):
         """Store a resource in the database"""
         resource = self.resource_class.from_orm(resource_create_instance)
         deserialize_resource_relationships(
             session, self.resource_class, resource, resource_create_instance
         )
         session.add(resource)
+        session.flush()
+
+        if user.is_connector:
+            # 1. Trust connector's input, do not override platform/platform_resource_identifier
+            # error message is raised if the connector does not provide these fields.
+            pass
```

with the per-platform roles, I would raise an error if the provided platform does not match the one the user/connector has permissions for

---

done